crates/oxc_linter/src/rules/jest/prefer_expect_resolves.rs

```diff
@@ -0,0 +1,267 @@
+use oxc_span::Span;
+
+use crate::{
+    context::LintContext,
+    fixer::Fix,
+    rule::Rule,
+    utils::{collect_possible_jest_call_node, parse_expect_jest_fn_call, PossibleJestNode},
+};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jest(prefer-expect-resolves): Prefer `await expect(...).resolves` over `expect(await ...)` syntax.")]
+#[diagnostic(severity(warning), help("Use `await expect(...).resolves` instead"))]
+struct ExpectResolves(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferExpectResolves;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// When working with promises, there are two primary ways you can test the resolved
+    /// value:
+    /// 1. use the `resolve` modifier on `expect`
+    /// (`await expect(...).resolves.<matcher>` style)
+    /// 2. `await` the promise and assert against its result
+    /// (`expect(await ...).<matcher>` style)
+    ///
+    /// While the second style is arguably less dependent on `jest`, if the promise
+    /// rejects it will be treated as a general error, resulting in less predictable
+    /// behaviour and output from `jest`.
+    ///
+    /// Additionally, favoring the first style ensures consistency with its `rejects`
+    /// counterpart, as there is no way of "awaiting" a rejection.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// // valid
+    /// it('passes', async () => {
+    ///     await expect(someValue()).resolves.toBe(true);
+    /// });
+    /// it('is true', async () => {
+    ///     const myPromise = Promise.resolve(true);
+    ///
+    ///     await expect(myPromise).resolves.toBe(true);
+    /// });
+    ///
+    /// it('errors', async () => {
+    ///     await expect(Promise.reject(new Error('oh noes!'))).rejects.toThrowError(
+    ///         'oh noes!',
+    ///     );
+    /// });
+    ///
+    /// // invalid
+    /// it('passes', async () => {
+    ///     expect(await someValue()).toBe(true);
+    /// });
+    /// it('is true', async () => {
+    ///     const myPromise = Promise.resolve(true);
+    ///     expect(await myPromise).toBe(true);
+    /// });
+    /// ```
+    PreferExpectResolves,
+    style,
+);
+
+impl Rule for PreferExpectResolves {
+    fn run_once(&self, ctx: &LintContext) {
+        for possible_jest_node in &collect_possible_jest_call_node(ctx) {
+            Self::run(possible_jest_node, ctx);
+        }
+    }
+}
+
+impl PreferExpectResolves {
+    fn run<'a>(possible_jest_node: &PossibleJestNode<'a, '_>, ctx: &LintContext<'a>) {
+        let node = possible_jest_node.node;
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        let Some(expect_fn_call) = parse_expect_jest_fn_call(call_expr, possible_jest_node, ctx)
+        else {
+            return;
+        };
+        let Some(parent) = expect_fn_call.head.parent else {
+            return;
+        };
+
+        if !parent.is_call_expression() {
```

Comment:
Seems useless because of the following check?

Reply:
Yes, I remove it.